tests/components/knx/test_climate.py

```diff
@@ -95,6 +89,10 @@ async def test_climate_on_off(
     # read on/off state
     await knx.assert_read("1/2/9")
     await knx.receive_response("1/2/9", 1)
+    # read heat/cool state
+    if heat_cool_ga:
```

Comment:
We should avoid using if statements in tests. Since this already was there i dont see it as blocker (because usually we should refactor tests for this, but that would make this PR too big)